engines/io_uring.c

```diff
@@ -554,6 +554,46 @@ static struct io_u *fio_ioring_cmd_event(struct thread_data *td, int event)
 	return io_u;
 }
 
+static char *fio_ioring_cmd_errdetails(struct io_u *io_u)
+{
+	unsigned int sct = (io_u->error >> 8) & 0x7;
+	unsigned int sc = io_u->error & 0xff;
+#define MAXERRDETAIL 1024
+#define MAXMSGCHUNK 128
+	char *msg, msgchunk[MAXMSGCHUNK];
+	int ctrlid, nsid;
+	int ret;
+
+	msg = calloc(1, MAXERRDETAIL);
+	strcpy(msg, "io_uring_cmd: ");
+
+	snprintf(msgchunk, MAXMSGCHUNK, "%s: ", io_u->file->file_name);
+	strlcat(msg, msgchunk, MAXERRDETAIL);
+
+	/*
+	 * Check the filename to figure out cmd_type since we don't have
+	 * thread_data here.
+	 */
+	ret = sscanf(io_u->file->file_name, "/dev/ng%dn%d", &ctrlid, &nsid);
+	if (ret == 2) {
+		strlcat(msg, "cq entry status (", MAXERRDETAIL);
+
+		snprintf(msgchunk, MAXMSGCHUNK, "sct=0x%02x; ", sct);
+		strlcat(msg, msgchunk, MAXERRDETAIL);
+
+		snprintf(msgchunk, MAXMSGCHUNK, "sc=0x%02x", sc);
+		strlcat(msg, msgchunk, MAXERRDETAIL);
+
+		strlcat(msg, ")", MAXERRDETAIL);
```

Comment:
Can you remove this `strlcat` and just add the `)` to the `snprtinf` call above this one?

Reply:
Okay!